app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,54 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, except: [:show, :index]
+# index / show the post index page
+  def index
+    @posts = Post.all
+  end
+#
+# new / for making new post part 1
+  def new
+    @posts = Post.new
+  end
+
+# create / for making new post part 2
+  def create
+    @posts = current_user.posts.create!(post_params)
+    # @posts = Post.create!(post_params)
+    redirect_to post_path(@posts)
+  end
+
+#show / for showing the post page
+  def show
+    @posts = Post.find(params[:id])
+  end
+
+# edit / edit a post part 1
+  def edit
+    @posts = Post.find(params[:id])
+    if current_user.id != @posts.user_id
+      redirect_to post_path(@posts)
+    end
+  end
+
+# update / edit a post part 2
+  def update
+    @posts = Post.find(params[:id])
+    @posts.update(post_params)
+
+    redirect_to post_path(@posts)
+  end
+
+# destroy/ delete a post
+  def destroy
+    @posts = Post.find(params[:id])
```

Comment:
How can we protect against unwanted deletion of posts? Who should be able to delete posts?